docs/states.md

```diff
@@ -0,0 +1,53 @@
+# Application States (#status)
+
+
+`ApplicationChoice` uses a State Machine to manage the states the model can be in.
+
+We use a library called [workflow](https://github.com/geekq/workflow) to manage this.
+
+The workflow definition is stored in [ApplicationStateChange](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/services/application_state_change.rb).
+
+
+## Current States
+
+This is a table of all the states that an ApplicationChoice can have.
+
+
+|State                         | Description|
+|---                           | ---|
+| `application_not_sent`       | Applications are set to this state at the end of the cycle if they are `unsubmitted` |
+| `awaiting_provider_decision` | Submitted and waiting for provider decision |
+| `cancelled`                  | DEPRECATED |
+| `conditions_not_met`         | Submitted, accepted but offer conditions were not met by the candidate |
+| `declined`                   | Offer by the provider is declined |
+| `inactive`                   | Application has been in `awaiting_provider_decision` for more than 30 days |
+| `interviewing`               | An interview is created by the provider for the candidate |
+| `offer`                      | Application has received an offer from the provider |
+| `offer_deferred`             | An offer has been deferred to next recruitement cycle |
+| `offer_withdrawn`            | Provider withdraws an offer |
+| `pending_conditions`         | Provider makes and offer and the caniddate must meet conditions of the offer |
+| `recruited`                  | Unconditional offer is accepted or offer conditions have been met |
+| `rejected`                   | Application is rejected by provider accompanied by structured reasons|
+| `unsubmitted`                | Candidate has added the course to their application form but has not submitted it yet |
+| `withdrawn`                  | Submitted application has been withdrawn by the candidate (or by the provider on behalf of the canidate) |
+
+## State Categories
+
+These are groupings of states that allow us to define behaviour for a any application whose status is in the group.
+
+|Category|States|Description|
+|---|----|---|
+|STATES_NOT_VISIBLE_TO_PROVIDER |unsubmitted<br>cancelled<br>application_not_sent|These states provide functionality in the system but are not exposed to the Providers. They are either hidden completely from the API or aliased to another state name|
+|STATES_VISIBLE_TO_PROVIDER|awaiting_provider_decision<br>interviewing<br>offer<br>pending_conditions<br>recruited<br>rejected<br>declined<br>withdrawn<br>conditions_not_met<br>offer_withdrawn<br>offer_deferred<br>inactive|Theoretically the opposite of VISIBLE_TO_PROVIDER|
```

Comment:
Is `inactive` visible to providers? I thought we show them something else. I'm not sure though, I could very well be wrong here.

Reply:
Yes, I meant "opposite of NOT_VISIBLE_TO_PROVIDERS" and also I will reference that we alias the status names in the Vendor API (and the ProviderInterface, I think)